demo_exam/tasks.h

```diff
@@ -0,0 +1,64 @@
+
+// DD.MM
+const int STR_LEN = 5, DOT_INDEX = 2;
+const int DAY_MIN = 1, DAY_MAX = 31;
+const int MONTH_MIN = 1, MONTH_MAX = 12;
+
+bool CheckDate(const char arr[]) {
+    if(arr == nullptr) {
+        return false;
+    }
+    for(int i = 0; i < STR_LEN; i++) {
+        if((i == DOT_INDEX && arr[i] != '.')) {
+            return false;
+        }
+        else if(i != DOT_INDEX && !(arr[i] >= '0' && arr[i] <= '9')) {
+            return false;
+        }
+    }
+    int day = (arr[0] - '0') * 10 + (arr[1] - '0'), month = (arr[3] - '0') * 10 + (arr[4] - '0');
+    if(!(day >= DAY_MIN && day <= DAY_MAX && month >= MONTH_MIN && month <= MONTH_MAX)) {
+        return false;
+    }
+    return true;
+}
+
+const double EPSILON = 0.00000000000000000000001;
+
+double abs(double num) {
+    return num < 0 ? -num : num;
+}
+
+bool CheckSquare(double x1, double y1, double x2, double y2) {
+//    return abs(x1) + abs(x2) == abs(y1) + abs(y2);
+    return abs(abs(x1) + abs(x2) - abs(y1) - abs(y2)) < EPSILON;

```

Comment:
Do you consider the case in which the two points defined by x1, y1 and x2,y2 respectively, are one and the same point? Can they be the opposites of a square's diagonal? 
Also, this check won't give you a correct answer. You can try with (-1, -2, -2, -3) and (1, 1, 3, 3) as parameters, it should be true. What is your logic behind the check and did you sketch it on paper first? :)

Reply:
I actually sketched myself an example, which turned out to be a beatiful edge case and resolved my logic around it. I should have tested more and probably would have found that it is quite wrong.
![square](https://user-images.githubusercontent.com/25901020/216170203-219ea3a7-df10-4292-846f-498e375d1101.png)